resilience4j-framework-common/src/test/java/io/github/resilience4j/common/circuitbreaker/configuration/CircuitBreakerConfigurationPropertiesTest.java

```diff
@@ -470,6 +470,17 @@ public void testIgnoreExceptionPredicateWithBaseConfig() {
         assertThat(circuitBreakerConfig.getIgnoreExceptionPredicate().test(new NullPointerException())).isFalse();
     }
 
+    @Test(expected = IllegalStateException.class)
+    public void testCircularReferenceInBaseConfigThrowsIllegalStateException() {
+        CommonCircuitBreakerConfigurationProperties circuitBreakerConfigurationProperties = new CommonCircuitBreakerConfigurationProperties();
+
+        CommonCircuitBreakerConfigurationProperties.InstanceProperties selfReferencingConfig = new CommonCircuitBreakerConfigurationProperties.InstanceProperties();
+        selfReferencingConfig.setBaseConfig("selfReferencingConfig");
+        circuitBreakerConfigurationProperties.getConfigs().put("selfReferencingConfig", selfReferencingConfig);
+
+        circuitBreakerConfigurationProperties.createCircuitBreakerConfig("selfReferencingConfig", selfReferencingConfig, compositeCircuitBreakerCustomizer());
+    }
+
```

Comment:
I think we should add a test for the specific reason this PR exists. The test `testCreateCircuitBreakerRegistryWithSharedConfigs` (line 157) does test base configs, but not in combination with the default configuration.

```java
@Test
    public void testCircuitBreakerConfigurationWithDefaultAndBaseConfig() {
        //Given
        CommonCircuitBreakerConfigurationProperties.InstanceProperties baseProperties = new CommonCircuitBreakerConfigurationProperties.InstanceProperties();
        baseProperties.setSlidingWindowType(CircuitBreakerConfig.SlidingWindowType.COUNT_BASED);
        baseProperties.setSlidingWindowSize(2000);
        baseProperties.setMaxWaitDurationInHalfOpenState(Duration.ofMillis(100L));
        CommonCircuitBreakerConfigurationProperties.InstanceProperties defaultProperties = new CommonCircuitBreakerConfigurationProperties.InstanceProperties();
        defaultProperties.setSlidingWindowType(CircuitBreakerConfig.SlidingWindowType.TIME_BASED);
        defaultProperties.setSlidingWindowSize(1000);
        defaultProperties.setBaseConfig("baseConfig");

        CommonCircuitBreakerConfigurationProperties circuitBreakerConfigurationProperties = new CommonCircuitBreakerConfigurationProperties();
        circuitBreakerConfigurationProperties.getConfigs().put("baseConfig", baseProperties);
        circuitBreakerConfigurationProperties.getConfigs().put("default", defaultProperties);

        //Then
        assertThat(circuitBreakerConfigurationProperties.getConfigs()).hasSize(2);
        var config = circuitBreakerConfigurationProperties.createCircuitBreakerConfig("default", defaultProperties, compositeCircuitBreakerCustomizer());

        // Should get default config and overwrite registerHealthIndicator and eventConsumerBufferSize but not allowHealthIndicatorToFail
        assertThat(config.getMaxWaitDurationInHalfOpenState()).isEqualTo(baseProperties.getMaxWaitDurationInHalfOpenState());
        assertThat(config.getSlidingWindowSize()).isEqualTo(defaultProperties.getSlidingWindowSize());
        assertThat(config.getSlidingWindowType()).isEqualTo(defaultProperties.getSlidingWindowType());
    }
```

Reply:
Will do, I will add a test for this situation specifically. Thanks for pointing it out!

Reply:
I've added the method that tests a base config in combination with the default configuration.